dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java

```diff
@@ -184,8 +187,47 @@ private String normalizePath(String basePath, String path) {
             return path.startsWith("/") ? basePath + path : basePath + "/" + path;
         }
 
-        public List<String> getEndpoints() {
-            return Ordering.natural().sortedCopy(endpoints);
+        public List<EndpointLogLine> getEndpointLogLines() {
+            return endpointLogLines;
+        }
+    }
+
+    private static class EndpointLogLine {
+        private final String httpMethod;
+        private final String basePath;
+        private final Class<?> klass;
+
+        public EndpointLogLine(String httpMethod, String basePath, Class<?> klass) {
+            this.basePath = basePath;
+            this.klass = klass;
+            this.httpMethod = httpMethod;
+        }
+
+        public String getHttpMethod() {
+            return httpMethod;
+        }
+
+        public String getBasePath() {
+            return basePath;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("    %-7s %s (%s)", httpMethod, basePath, klass.getCanonicalName());
+        }
+    }
+
+    private static class EndpointComparator implements Comparator<EndpointLogLine>, Serializable {
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public int compare(EndpointLogLine endpointA, EndpointLogLine endpointB) {
+            int basePathComparison = endpointA.getBasePath().compareTo(endpointB.getBasePath());
```

Comment:
Thanks for tip, didn't know about it I'll check it out!
On Apr 20, 2015 9:35 PM, "Artem Prigoda" <notifications@github.com> wrote:

> In
> dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java
> <https://github.com/dropwizard/dropwizard/pull/1002#discussion_r28721112>:
>
> > +        public String getBasePath() {
> > +            return basePath;
> > +        }
> > +
> > +        @Override
> > +        public String toString() {
> > +            return String.format("    %-7s %s (%s)", httpMethod, basePath, klass.getCanonicalName());
> > +        }
> > +    }
> > +
> > +    private static class EndpointComparator implements Comparator<EndpointLogLine>, Serializable {
> > +        private static final long serialVersionUID = 1L;
> > +
> > +        @Override
> > +        public int compare(EndpointLogLine endpointA, EndpointLogLine endpointB) {
> > +            int basePathComparison = endpointA.getBasePath().compareTo(endpointB.getBasePath());
>
> What do you think about using Guava's ComparisonChain
> <https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ComparisonChain.java>
> ?
> Something like this:
>
> return ComparisonChain.start()
>                     .compare(this.basePath, other.basePath)
>                     .compare(this.httpMethod, other.httpMethod)
>                     .result();
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dropwizard/dropwizard/pull/1002/files#r28721112>.
>